chainntnfs/txnotifier.go

```diff
@@ -679,9 +679,12 @@ func (n *TxNotifier) RegisterConf(txid *chainhash.Hash, pkScript []byte,
 			confDetails = &confDetailsCopy
 		}
 
-		err := n.dispatchConfDetails(ntfn, confDetails)
-		if err != nil {
-			return nil, err
+		// Deliver the details to the conf set.
+		for _, ntfn := range confSet.ntfns {
```

Comment:
nit: don't shadow the outer `ntfn` name to make it more clear what notification we're notifying on?

Also, didn't look super deep into this, but could this lead to duplicate notifications? Probably not a big issue as we should always be able to handle duplicates.

Reply:
Or is there only ever one notification being in rescan? My assumption is multiple of the set could be and then they would all notify. Perhaps I'm also misunderstanding things here.

Reply:
> nit: don't shadow the outer ntfn name to make it more clear what notification we're notifying on?

Done.

> Or is there only ever one notification being in rescan? My assumption is multiple of the set could be and then they would all notify. Perhaps I'm also misunderstanding things here.

Yeah the water is a bit deep here, also not very confident about this package either😂 - I think the rescan is for every notification, which can def be optimized. As for duplicates, because the `ntfn` will be marked as `dispatched=true` and we will decide to notify or not by checking `dispatched`, I don't think we will see duplicates.

Reply:
turns out the `dispatched=true` only protects the `Confirmed` notification. For the `Updates` notification, which sends the num of confs left (I think we never use it?), we may still have duplicates. To avoid the dups, I created a commit in the end to skip if the same update has already been sent before.